apps/server/minecraft_manager/audit/repository.py

```
@@ -0,0 +1,179 @@
+
+        where = ("WHERE " + " AND ".join(conditions)) if conditions else ""
+
+        conn = sqlite3.connect(self._database)
+        try:
+            total: int = conn.execute(
+                f"SELECT COUNT(*) FROM audit_log {where}", params
+            ).fetchone()[0]
+
+            rows = conn.execute(
+                f"SELECT id, occurred_at, actor_identity, action, target, result, details "
+                f"FROM audit_log {where} ORDER BY occurred_at DESC LIMIT ? OFFSET ?",
```

Ruff (0.16.3)

[error] Possible SQL injection vector through string-based query construction

(S608)